### PR TITLE
fix(curriculum): tests passing with seed code in Role playing game

### PR DESCRIPTION
**Pull request by gikf**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

- - -
Notes
- Step 1 - browser appears to fix many of the incorrectly defined `html`, `head`, `body`, including the lack of them.
- Step 18 & 29, wanna see _funny_ trick?
  1. Go to any page (not fCC limited, youtube could be used as well), for example https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-18
  2. Open browser's console.
  3. Type either in browser's console, or in the editor: `console.log(button2)`
  4. See appear the `#button2` element in console.
  5. Try it with other ids with names that could be proper JS variables.
  6. Also see their elements printed in console.
  7. You can be as confused as I am.
- Steps with `pick` function and `if` statements. `pick.toString()` will added infinite loop protection to the loops. This uses `if`, which makes using `pick.toString()` to match `if` less convenient.

### CODE REVIEW

**Review by moT01 — CHANGES_REQUESTED** [review]
Looking good @gikf 👍 

I think you changed the tests on 18 and 29 cause of the funny trick you found. Here's what GPT says:

“In HTML, when you give an element an id, that id becomes a global variable accessible in the JavaScript environment. This is why you can reference button2 directly in your script without declaring a variable.”

But the first test on those steps checks for the variable declaration - e.g. `const button2` which overwrites the global. So I didn't see why they needed to be changed.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a7bf06d2ad9d1c5024e833.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ In your `goStore()` function, update the `onclick` property for each button to r
 You should use dot notation to access the `onclick` property of `button1`.
 
 ```js
-assert.match(code, /button1\.onclick/);
+assert.match(goStore.toString(), /button1\.onclick/);
```
[the remark this answers is not in the document]
Comment (gikf) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a7bf06d2ad9d1c5024e833.md:19]: In current form this test is not really checking anything - it matches seed code.

What do you think about adding to the hint text something like: _in your `goStore` function_?

**Comment by gikf** [comment]
> I think you changed the tests on 18 and 29 cause of the funny trick you found. Here's what GPT says:

That's the one!

> But the first test on those steps checks for the variable declaration - e.g. `const button2` which overwrites the global. So I didn't see why they needed to be changed.

It looks slightly like the best case scenario. The worst one is a bit nitpicking, but other test:
```js
assert.match(code, /const\s+button2/);
```
Would of course match `const button2something`. Only related test is changed, but it increases confidence for this one as well.

**Review by moT01 — APPROVED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a1166ed9a56d439c0770e7.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ assert.equal(spans?.length, 3);
 You should give the new three `span` elements a class of `stat`.
 
 ```js
-assert.exists(document.querySelectorAll('#stats > .stat')?.length, 3);
+assert.equal(document.querySelectorAll('#stats > .stat')?.length, 3);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a1166ed9a56d439c0770e7.md:24]:
```suggestion
assert.lengthOf(document.querySelectorAll('#stats > .stat'), 3);
```

**Review by majestic-owl448 — APPROVED** [review]

**Pull request merged**